src/Twilio/Security/RequestValidator.cs

```diff
@@ -124,23 +125,33 @@ private static bool SecureCompare(string a, string b)
             return mismatch == 0;
         }
 
-        private string RemovePort(UriBuilder uri)
+        private string RemovePort(string url)
         {
-            // UriBuilder.ToString() will not display the port 
-            // if the Port property is set to -1
+            var uri = new UriBuilder(url);
+            uri.Host = PreserveHostnameCasing(url);
```

Comment:
Why is this needed if `OriginalString` doesn't change the casing?

Reply:
Turns out that it still lower cases the host. Please see the comment above to Elise for details.

Reply:
What about other parts of the URL, like the scheme or credentials?

Reply:
Those are preserved, I'm only modifying the host. I will add some tests to verify.

Reply:
Added a test case that has credentials in the URL. Schemes were already being tested along with ports.

Reply:
Don't see any tests with an uppercase scheme or creds.

Reply:
Ah, I see. 👀

Reply:
Just added a few more tests. One with upper cased scheme and mixed case credentials.

Reply:
Turns out the credentials were preserved, but the scheme was not. Also needed to add logic to account for the existence of credentials when trying to extract the host.